### PR TITLE
feat: added support for JUnit Theory tests

### PR DESCRIPTION
**Pull request by asadsalman**
[JUnit Theories](https://junit.org/junit4/javadoc/4.12/org/junit/experimental/theories/Theories.html) were not parsed by Flank perviously. This PR adds support for them.

Added test.

## Checklist

- [ ] Documented
- [x] Unit tested
- [ ] Integration tests updated

### CODE REVIEW

File: test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt [hunk 1]
```diff
@@ -118,7 +118,8 @@ internal fun InstrumentationTestContext.getFlankTestMethods(
     testFilter: TestFilter
 ): List<FlankTestMethod> =
     getParametrizedClasses().let { parameterizedClasses: List<TestMethod> ->
-        DexParser.findTestMethods(test.local).asSequence()
+        DexParser.findTestMethods(test.local, listOf("org.junit.experimental.theories.Theory"))
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt:121]: Let's have this list as a separate value, we might want to extend it in the future.

WDYT?

**Review by pawelpasterz — APPROVED** [review]
Looking good, just small change request 👍🏻

File: test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt (hunk 1 above)
Comment (asadsalman) [reply test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt:121]: Do you mean as a parameter a user can pass or just a global const or something?

Comment (pawelpasterz) [reply test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt:121]: Private value within `CreateAndroidTestContext.kt` file

**Comment by pawelpasterz** [comment]
Follow-up: create IT for this case
cc: @axelzuziak-gogo

File: test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt (hunk 1 above)
Comment (Sloox) [reply test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt:121]: Agreed move it out then we can merge 👍

**Comment by Sloox** [comment]
@flank-it

File: test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt (hunk 1 above)
Comment (asadsalman) [reply test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt]: Done!

**Review by piotradamczyk5 — APPROVED** [review]

**Pull request merged**